scripts/dev/tidy.php

```diff
@@ -0,0 +1,366 @@
+    line endings the prevailing EOL (end of line) character is used as a
+    final new line.
+
+    When used on the php-src root directory it ignores the predefined set
+    of binary files and certain test files.
+
+    By default, original files are backed up (file.ext~).
+
+  OPTIONS:
+
+    -h, --help           Display this help
+        --no-whitespace  Disable trimming trailing whitespace
+        --no-eol         Disable fixing final end of lines
+        --no-colors      Disable colors in the script output
+        --no-backup      Disable backed up files. Use with caution to not
+                         overwrite something by accident.
+    -f, --fix            Edit and save files in place
+    -q, --quiet          Do not output any message
+
+  EXAMPLES:
+
+    Display help: php tidy.php -h
+
+    Check php-src: scripts/dev/tidy.php .
+
+    Fix php-src: scripts/dev/tidy.php --fix --no-backup .
+
+    Fix custom path or file: ./tidy.php --fix input-path
+
+END;
+}
+
+/**
+ * Trim trailing spaces and tabs from each line.
+ */
+function trimTrailingWhitespace(string $content): string
+{
+    return preg_replace('/[\t ]+(\r?$)/m', '$1', $content);
+}
+
+/**
+ * Trim final new lines and append an EOL (end of line) based on the previous
+ * prevailing EOL type used LF (\n), CRLF (\r\n) or CR (\r).
+ */
+function trimFinalEol(string $content): string
+{
+    $trimmed = rtrim($content, " \t\r\n");
+    $trimmed .= getEol($trimmed);
+
+    return $trimmed;
+}
+
+/**
+ * Get end of line based on the prevailing LF, CRLF or CR new line characters.
+ */
+function getEol(string $content): ?string
+{
+    // Empty files don't need a new line attached
+    if (empty($content)) {
+        return null;
+    }
+
+    // Match all LF, CRLF and CR EOL characters
+    preg_match_all('/(*BSR_ANYCRLF)\R/', $content, $matches);
+
+    // For single line files LF is returned
+    if (empty($matches[0])) {
+        return "\n";
+    }
+
+    $counts = array_count_values($matches[0]);
+    arsort($counts);
+    $prevailingEol = key($counts);
+
+    return $prevailingEol;
+}
+
+$fix = false;
+$whitespace = true;
+$eol = true;
+$colors = true;
+$backup = true;
+$quiet = false;
+
+foreach ($argv as $argument) {
+    switch ($argument) {
+        case ($argc < 2):
+        case '-h':
+        case '--help':
+            echo usage();
+            exit;
+        break;
+        case '--no-whitespace':
+            $whitespace = false;
+        break;
+        case '--no-eol':
+            $eol = false;
+        break;
+        case '-f':
+        case '--fix':
+            $fix = true;
+        break;
+        case '--no-colors':
+            $colors = false;
+        break;
+        case '--no-backup':
+            $backup = false;
+        break;
+        case '-q':
+        case '--quiet':
+            $quiet = true;
+        break;
+        case (file_exists($argument) && $argument !== $argv[0]):
```

Comment:
This one should also be inside the default case ^^

Reply:
Fixed.